src/test/java/org/primeframework/mvc/test/RequestBuilder.java

```diff
@@ -832,18 +833,29 @@ HTTPResponseWrapper run() {
 
     // Now that the cookies are ready, if the CSRF token is enabled and the parameter isn't set, we set it to be consistent
     // since the [@control.form] would normally set that into the form and into the request.
-    MVCConfiguration configuration = injector.getInstance(MVCConfiguration.class);
-    if (configuration.csrfEnabled()) {
-      CSRFProvider csrfProvider = injector.getInstance(CSRFProvider.class);
-      if (csrfProvider.getTokenFromRequest(request) == null) {
-        String token = csrfProvider.getToken(request);
-        if (token != null) {
-          String parameterName = csrfProvider.getParameterName();
-          request.addURLParameter(parameterName, token);
+    if (request.getMethod() == HTTPMethod.POST) {
```

Comment:
This is a change to the current behavior to only append this when this is a `POST` and to add it to the request body as a `form` post would do. 

I don't expect this to cause any problems? Always adding it - even on a `GET` - and adding it as a query parameter is not ideal for some testing scenarios.

Reply:
Test?

Reply:
This is implicitly tested via FusionAuth, but good point - I added a test to assert this change.